tests/test_inference.py

```diff
@@ -2508,6 +2518,228 @@ def test_many_trees(self):
+        assert ts.equals(ts_postprocessed, ignore_provenance=True)
+
+    def test_has_shortened_edge_over_grand_root(self, small_sd_fixture):
+        # If there is something like a virtual-root-like ancestor, but
+        # which doesn't extend over the whole genome, we don't remove it
+        ts = tsinfer.infer(small_sd_fixture, post_process=False)
+        tables = ts.dump_tables()
+        tables.edges[ts.num_edges - 1] = tables.edges[ts.num_edges - 1].replace(left=1)
+        assert not tsinfer.has_single_edge_over_grand_root(tables.tree_sequence())
+
+    def test_virtual_like_root_removed(self, medium_sd_fixture):
+        ts = tsinfer.infer(medium_sd_fixture, post_process=False)
+        ts_simplified = ts.simplify(keep_unary=True)
+        assert tsinfer.has_single_edge_over_grand_root(ts)
+        ts_post_processed = tsinfer.post_process(ts, split_mrca=False)
+        assert ts_post_processed.num_edges == ts_simplified.num_edges - 1
+        assert not tsinfer.has_single_edge_over_grand_root(ts_post_processed)
+
+    def test_dont_split_edges_twice(self, medium_sd_fixture):
+        ts = tsinfer.infer(medium_sd_fixture, post_process=False)
+        ts = tsinfer.post_process(ts, split_mrca=False)
+        assert tsinfer.has_same_root_everywhere(ts)
+        # Once the mrca has been split, it can't be split again
+        tables = ts.dump_tables()
+        tsinfer.split_grand_mrca(tables)
+        ts = tables.tree_sequence()
+        # should fail has_same_root_everywhere, so will not be split again
+        assert not tsinfer.has_same_root_everywhere(ts)
+
+    def test_sample_order(self, medium_sd_fixture):
+        anc = tsinfer.generate_ancestors(medium_sd_fixture)
+        ats = tsinfer.match_ancestors(medium_sd_fixture, anc)
+        idx = [0, medium_sd_fixture.num_samples // 2, medium_sd_fixture.num_samples - 1]
+        ts = tsinfer.match_samples(
+            medium_sd_fixture, ats, indexes=idx, post_process=False
+        )
+        for i, (orig_id, inferred_node_id) in enumerate(zip(idx, ts.samples())):
+            assert i != inferred_node_id
+            sd_individual = medium_sd_fixture.individual(
+                medium_sd_fixture.sample(orig_id).individual
+            )
+            inferred_individual = ts.individual(ts.node(inferred_node_id).individual)
+            assert sd_individual.metadata["id"] == inferred_individual.metadata["id"]
+        ts = tsinfer.post_process(ts)
+        # After post-processing, samples should be 0..n
+        for i, (orig_id, inferred_node_id) in enumerate(zip(idx, ts.samples())):
+            assert i == inferred_node_id
+            sd_individual = medium_sd_fixture.individual(
+                medium_sd_fixture.sample(orig_id).individual
+            )
+            inferred_individual = ts.individual(ts.node(inferred_node_id).individual)
+            assert sd_individual.metadata["id"] == inferred_individual.metadata["id"]
+
```

Comment:
> We need a test here for the case where we _infer_ a single tree (recombination rate = ~0), verifying that we don't change the input at all. 

Sorry, I'm being dense and don't quite understand your comments here. You mean that we don't change the genotypes, or that we infer exactly the same tree as we used to generate the variant data. If you mean the same tree, then we do need to drop the oldest edge (as that won't be in the original, simulated tree)

Reply:
Thanks for clarifying just now @jeromekelleher  - I've added a check so that we simply return without doing anything if there is only a single edge to the grand MRCA, so we don't delete and recreate a node.